src/clr-angular/data/datagrid/providers/column-reorder.service.spec.ts

```diff
@@ -0,0 +1,180 @@
+        this.firstVisibleChecked = true;
+      }),
+      this.columnsService.checkLastVisible.subscribe(() => {
+        this.lastVisibleChecked = true;
+      })
+    );
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.forEach(sub => sub.unsubscribe());
+  }
+}
+
+export default function(): void {
+  describe('Column Reorder Service', function() {
+    let fixture: ComponentFixture<any>;
+
+    let testComponent: TestComponent;
+    let columnReorderService: ColumnReorderService;
+    let columnsService: MockColumnsService;
+
+    beforeEach(function() {
+      TestBed.configureTestingModule({
+        declarations: [TestComponent],
+        providers: [MOCK_COLUMN_SERVICE_PROVIDER, ColumnReorderService],
+      });
+      fixture = TestBed.createComponent(TestComponent);
+      fixture.detectChanges();
+      columnReorderService = TestBed.get(ColumnReorderService);
+      columnsService = <MockColumnsService>TestBed.get(ColumnsService);
+      columnsService.mockColumns(3);
+
+      testComponent = fixture.componentInstance;
+    });
+
+    afterEach(() => {
+      fixture.destroy();
+    });
+
+    it('provides the unique string id for column group', function() {
+      expect(columnReorderService.columnsGroupId).toBeString();
+    });
+
+    it('provides the view container', function() {
+      expect(columnReorderService.containerRef).not.toBeUndefined();
+    });
+
+    it('emits reorder request data when reorderViews method gets called', function() {
+      expect(testComponent.reorderRequest).toBeUndefined();
+      columnReorderService.reorderViews(testComponent.view1, testComponent.view2, null);
+      expect(testComponent.reorderRequest).not.toBeUndefined();
+    });
+
+    it('updates orders property value through updateOrders method', function() {
+      expect(columnReorderService.orders).toBeUndefined();
+      columnReorderService.updateOrders([1, 0, 2]);
+      expect(columnReorderService.orders).toEqual([1, 0, 2]);
+    });
+
+    it('updates column states on orders', function() {
+      expect(columnsService.columns[0].value.order).toBeUndefined();
+      expect(columnsService.columns[1].value.order).toBeUndefined();
+      expect(columnsService.columns[2].value.order).toBeUndefined();
+
+      columnReorderService.updateOrders([1, 0, 2]);
+
+      expect(columnsService.columns[0].value.order).toBe(1);
+      expect(columnsService.columns[1].value.order).toBe(0);
+      expect(columnsService.columns[2].value.order).toBe(2);
+    });
+
+    it('requests first and last visible columns checks after updating orders', function() {
+      expect(testComponent.firstVisibleChecked).toBeFalsy();
+      expect(testComponent.lastVisibleChecked).toBeFalsy();
+      columnReorderService.updateOrders([1, 0, 2]);
+      expect(testComponent.firstVisibleChecked).toBeTruthy();
+      expect(testComponent.lastVisibleChecked).toBeTruthy();
+    });
+
+    it('returns new order at index', function() {
```

Comment:
nit: should this be phrased as `orderAt` ?

Reply:
I think it should stay as `order at` because it's a new allocated order at that particular index.